store/types/gas.go

```diff
@@ -52,26 +53,37 @@ type GasMeter interface {
 type basicGasMeter struct {
 	limit    Gas
 	consumed Gas
+	lock *sync.Mutex
```

Comment:
BlockGas can/will be modified concurrently

Reply:
would it make sense to make Gas an atomic by using
`atomic.AddUint64`  and `atomic. AddUint64`? This way if we add / remove any functions that touch gas, we know that the locking is already handled at a lower layer. O/w we need to be careful of ensuring every function that accesses any `Gas` variable handles the mutex \cc @codchen . Not a blocker, though